src/control/AudioController.cpp

```diff
@@ -16,8 +18,8 @@ auto AudioController::startRecording() -> bool {
         time_t secs = time(nullptr);
         tm* t = localtime(&secs);
         // This prints the date and time in ISO format.
-        sprintf(buffer.data(), "%04d-%02d-%02d_%02d-%02d-%02d", t->tm_year + 1900, t->tm_mon + 1, t->tm_mday,
-                t->tm_hour, t->tm_min, t->tm_sec);
+        snprintf(buffer.data(), buffer.size(), "%04d-%02d-%02d_%02d-%02d-%02d", t->tm_year + 1900, t->tm_mon + 1,
+                 t->tm_mday, t->tm_hour, t->tm_min, t->tm_sec);
```

Comment:
Try to use `std::get_time` and `std::put_time` in c++17 for time conversions.

Reply:
I do not have the knowledge to do this change unfortunately. I have no understanding of locale issues and what `AudioController` does. I am confident that the `snprintf` change is equivalent, but I would not be so confident using `get_time` and `put_time` myself.